core/src/main/scala/offheap/Array.scala

```diff
@@ -23,6 +23,8 @@ final class Array[A] private (val addr: Addr) extends AnyVal {
   def transform(f: A => A): Array[A]                     = macro macros.ArrayApi.transform
   def filter(f: A => Boolean)
             (implicit a: Allocator): Array[A]            = macro macros.ArrayApi.filter
+  def foldLeft[B](z: B, op: (B, A) => B): B              = macro macros.ArrayApi.foldLeft[B]
```

Comment:
For consistency with standard collections lets use compatible signature here:
```
def foldLeft[B](z: B)(op: (B, A) ⇒ B): B
```
Having it all in the same argument list might can be quite confusing.